watertap/unit_models/tests/test_uv_aop.py

```diff
@@ -741,8 +732,242 @@ def test_costing(self, UV_frame):
 
         # Check solutions
-        assert pytest.approx(1544038, rel=1e-5) == value(m.fs.unit.costing.capital_cost)
+        assert pytest.approx(1767152, rel=1e-5) == value(m.fs.unit.costing.capital_cost)
         assert pytest.approx(90400.7, rel=1e-5) == value(
             m.fs.unit.costing.fixed_operating_cost
         )
-        assert pytest.approx(0.0192355, rel=1e-5) == value(m.fs.costing.LCOW)
+        assert pytest.approx(0.02023034, rel=1e-5) == value(m.fs.costing.LCOW)
+
+
+class TestUV_detailed:
+    @pytest.fixture(scope="class")
+    def UV_frame(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.properties = props.NDMAParameterBlock()
+
+        m.fs.unit = Ultraviolet0D(
+            default={
+                "property_package": m.fs.properties,
+                "uv_dose_type": UVDoseType.calculated,
+            }
+        )
+
+        # Example system for verifying costing
+        feed_flow_mass = 1026.5 * pyunits.kg / pyunits.s
+        feed_mass_frac_NDMA = 74e-9
+        feed_pressure = 101325 * pyunits.Pa
+        feed_temperature = (273.15 + 25) * pyunits.K
+        uv_intensity = 1 * pyunits.mW / pyunits.cm**2
+        # exporure_time = 32 * pyunits.s
+        inactivation_rate = 180 * pyunits.cm**2 / pyunits.J
+        reaction_rate_constant = 0 * pyunits.min**-1
+        EEO = 0.0259 * pyunits.kWh / pyunits.m**3
+        lamp_efficiency = 0.8
+        UVT = 0.9
+
+        feed_mass_frac_H2O = 1 - feed_mass_frac_NDMA
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "NDMA"].fix(
+            feed_flow_mass * feed_mass_frac_NDMA
+        )
+
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(
+            feed_flow_mass * feed_mass_frac_H2O
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1e-3, index=("Liq", "H2O")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1e5, index=("Liq", "NDMA")
+        )
+
+        m.fs.unit.inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.uv_intensity.fix(uv_intensity)
+        m.fs.unit.inactivation_rate["Liq", "NDMA"].fix(inactivation_rate)
+        m.fs.unit.reaction_rate_constant["Liq", "NDMA"].fix(reaction_rate_constant)
+        m.fs.unit.outlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.electrical_efficiency_phase_comp[0, "Liq", "NDMA"].fix(EEO)
+        m.fs.unit.lamp_efficiency.fix(lamp_efficiency)
+
+        # UV dose specifications
+        m.fs.unit.A_coeff.fix(2.15138466202152)
+        m.fs.unit.B_coeff.fix(10.2071744508813)
+        m.fs.unit.C_coeff.fix(0.696709253161921)
+        m.fs.unit.D_coeff.fix(1.09563014019707)
+        m.fs.unit.relative_lamp_output.fix(12)
+        m.fs.unit.num_of_banks.fix(2)
+        m.fs.unit.UVT.fix(UVT)
+        return m
+
+    @pytest.mark.unit
+    def test_config(self, UV_frame):
```

Comment:
I think you should add assertions for the UVDoseType option and any other config options in the model.

Reply:
Added